addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -133,6 +133,9 @@ type ObjectReference struct {
 	// +kubebuilder:validation:Required
 	// +required
 	Resource string `json:"resource"`
+	// namespace of the referent.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
why do you need this, it is supposed to be the same ns of managedclusteraddon?

Reply:
yes, our controller knows it should be in the same namespace, but for namespace scoped resource, `must-gather` will fail:
```
oc adm inspect managedclusteraddons.addon.open-cluster-management.io  --all-namespaces  --dest-dir=must-gather
...
error: errors ocurred while gathering data:
    skipping gathering klusterletaddonconfigs.agent.open-cluster-management.io/local-cluster due to error: a resource cannot be retrieved by name across all namespace
```

tried with namespace still the same
```
oc adm inspect managedclusteraddons.addon.open-cluster-management.io  -n local-cluster  --dest-dir=must-gather
error: errors ocurred while gathering data:
    skipping gathering klusterletaddonconfigs.agent.open-cluster-management.io/local-cluster due to error: a resource cannot be retrieved by name across all namespaces
```

Reply:
ah, ok